### DISCUSSION TITLE
Is it good practise to run migration each I boot my server?

### DISCUSSION THREAD

**Discussion by jjauzion** [golang-migrate/migrate#701] [Q&A] [answered]
I'm kind of new to SQL migration and I really like this package !! 

I was wondering how best to run migration on my prod server. I'm using SQLite3 so I used the Go API as suggested in #670 and I have a MigrateUp() and MigrateDown() function.
At the moment I just call the MigrateUp() function every time my server start and just log the error in case the function return with error. This way I don't have to manually run a command on the server every time I add a migration to my DB.

But I was wondering if this was safe to run the MigartionUp() every time ?

**Comment by dhui** [comment, marked as answer]
For SQLite, probably. e.g. assuming you only have one server
There might be issues if you start multiple instances of your application since the sqlite3 db driver [doesn't support](https://github.com/golang-migrate/migrate/blob/master/database/sqlite3/sqlite3.go#L181) distributed locks

Also if you need to rollback migrations (assuming doing so is compatible with your application) or a migration fails, you'll need to use the CLI. For failed migrations, you'll need to fix the issue and force the dirty migration to the previous version.

**Reply by jjauzion** [reply]
ok thx for helping :)
